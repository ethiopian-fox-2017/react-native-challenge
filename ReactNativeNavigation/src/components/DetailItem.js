import React from 'react'
import {
  View,
  Text,
  Image,
  TouchableHighlight,
  ScrollView,
  CameraRoll,
  Platform,
  Dimensions,
  Alert
} from 'react-native';
import { Card } from 'react-native-elements';
import RNFetchBlob from 'react-native-fetch-blob';


const deviceWidth = Dimensions.get('window').width;
const deviceHeight = Dimensions.get('window').height;

const styles = {
  itemStyle: {
    flexDirection: 'column',
    alignItems: 'center'
  },
  imageStyle: {
    width: deviceWidth,
    height: deviceHeight * 0.6
  },
  detailStyle: {
    fontWeight: 'bold',
    fontSize: 15,
    color: '#66BB6A'
  },
  saveButton: {
    alignSelf: 'center',
    color: 'pink',
    borderWidth: 2,
    borderRadius: 10,
    borderColor: 'pink',
    padding: 10,
    margin: 10,
  },
  cardStyle: {
    alignItems: 'center'
  },
  fullWidth: {
    width: '100%'
  }
}


class DetailItem extends React.Component {

  constructor(props){
    super(props);

  }

  static navigationOptions = { title: 'Detail' };

  // handleSaveImage(){
  //   let uri = this.props.navigation.state.params.item.image_url[1];
  //   console.log('saveImage',uri);
  //   let promise = CameraRoll.saveImageWithTag(uri);
  //   promise.then(function(result) {
  //     console.log('save succeeded ' + result);
  //   }).catch(function(error) {
  //     console.log('save failed ' + error);
  //   });
  // }

  saveToCameraRoll = () => {
    let uri = this.props.navigation.state.params.item.image_url[1];
    if (Platform.OS === 'android') {
      RNFetchBlob
      .config({
        fileCache : true,
        appendExt : 'jpg'
      })
      .fetch('GET', uri)
      .then((res) => {
        CameraRoll.saveToCameraRoll(res.path())
          .then(Alert.alert('Success', 'Photo added to camera roll!'))
          .catch(err => console.log('err:', err))
      })
    } else {
      CameraRoll.saveToCameraRoll(uri)
        .then(Alert.alert('Success', 'Photo added to camera roll!'))
    }
  }

  render() {
    const { params } = this.props.navigation.state;
    const title = params.item.id + ' | '+ params.item.name;
    return (
      <View style={styles.itemStyle}>
          <Image style={styles.imageStyle} source={{uri: params.item.image_url[1]}} />

        <ScrollView style={styles.fullWidth}>
        <Card style={styles.cardStyle}
          title={title}>
          <Text style={styles.detailStyle}> Camera : {params.item.camera ? params.item.camera : '-'} </Text>
          <Text style={styles.detailStyle}> Lens : {params.item.lens ? params.item.lens : '-'} </Text>
          <Text style={styles.detailStyle}> Photo By : {params.item.user.firstname} {params.item.user.lastname} </Text>
          <TouchableHighlight onPress={()=>this.saveToCameraRoll()}>
            <Text style={styles.saveButton}>Save</Text>
          </TouchableHighlight>
        </Card>
        </ScrollView>
      </View>

    )
  }
}


export default DetailItem;
