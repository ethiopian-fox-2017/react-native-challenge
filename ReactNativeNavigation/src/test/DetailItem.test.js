import renderer from 'react-test-renderer';
import React from 'react';
import ReactDOM from 'react-dom';

import DetailItem from '../components/DetailItem';


it('render and match to snapshot', () => {
  const dummyItem = {
    id: 0,
    name: '',
    image_url: [],
    camera: '',
    lens: '',
    user:{
      firstname: '',
      lastname: ''
    }
  }
  const navigation = {
    state: {
      params : {
        item: dummyItem
      }
    }
  }
  const tree = renderer.create(
      <DetailItem navigation={navigation} />
  )
  expect(tree).toMatchSnapshot();
})
