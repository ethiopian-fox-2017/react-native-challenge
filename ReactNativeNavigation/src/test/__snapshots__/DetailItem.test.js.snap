// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render and match to snapshot 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "column",
    }
  }
>
  <Image
    source={
      Object {
        "uri": undefined,
      }
    }
    style={
      Object {
        "height": "70%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "white",
          "borderColor": "#e1e8ee",
          "borderWidth": 1,
          "margin": 15,
          "marginBottom": 0,
          "padding": 15,
          "shadowColor": "rgba(0,0,0, .2)",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 1,
          "shadowRadius": 1,
        },
        undefined,
        undefined,
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "backgroundColor": "transparent",
          },
          undefined,
          undefined,
        ]
      }
    >
      <View>
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {},
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              undefined,
              Array [
                Object {
                  "color": "#43484d",
                  "fontSize": 17.5,
                  "fontWeight": "bold",
                  "marginBottom": 15,
                  "textAlign": "center",
                },
                undefined,
                undefined,
                undefined,
              ],
            ]
          }
        >
          0 | 
        </Text>
        <View
          style={
            Array [
              Object {
                "backgroundColor": "#e1e8ee",
                "height": 1,
              },
              Array [
                Object {
                  "marginBottom": 15,
                },
                undefined,
              ],
            ]
          }
        />
      </View>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#66BB6A",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
         Camera : 
        -
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#66BB6A",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
         Lens : 
        -
         
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#66BB6A",
            "fontSize": 15,
            "fontWeight": "bold",
          }
        }
      >
         Photo By : 
        
         
        
         
      </Text>
    </View>
  </View>
</View>
`;
